DigitalLearningSolutions.Data/Services/CourseAdminFieldsService.cs

```diff
@@ -0,0 +1,217 @@
+        private readonly IUserDataService userDataService;
+
+        public CourseAdminFieldsService(
+            ICourseAdminFieldsDataService courseAdminFieldsDataService,
+            ILogger<CourseAdminFieldsService> logger,
+            IUserDataService userDataService
+        )
+        {
+            this.courseAdminFieldsDataService = courseAdminFieldsDataService;
+            this.logger = logger;
+            this.userDataService = userDataService;
+        }
+
+        public CourseAdminFields? GetCustomPromptsForCourse(
+            int customisationId,
+            int centreId,
+            int categoryId = 0
+        )
+        {
+            var result = courseAdminFieldsDataService.GetCourseAdminFields(customisationId, centreId, categoryId);
+            return new CourseAdminFields(
+                customisationId,
+                centreId,
+                PopulateCustomPromptListFromCourseCustomPromptsResult(result)
+            );
+        }
+
+        public List<CustomPromptWithAnswer> GetCustomPromptsWithAnswersForCourse(
+            DelegateCourseInfo delegateCourseInfo,
+            int customisationId,
+            int centreId,
+            int categoryId = 0
+        )
+        {
+            var result = GetCourseCustomPromptsResultForCourse(customisationId, centreId, categoryId);
+
+            return PopulateCustomPromptWithAnswerListFromCourseAdminFieldsResult(result, delegateCourseInfo);
+        }
+
+        public void UpdateCustomPromptForCourse(int customisationId, int promptNumber, bool mandatory, string? options)
+        {
+            courseAdminFieldsDataService.UpdateCustomPromptForCourse(customisationId, promptNumber, mandatory, options);
+        }
+
+        public void RemoveCustomPromptFromCourse(int customisationId, int promptNumber)
+        {
+            using var transaction = new TransactionScope();
+            try
+            {
+                userDataService.DeleteAllAnswersForAdminField(customisationId, promptNumber);
+                courseAdminFieldsDataService.UpdateCustomPromptForCourse(
+                    customisationId,
+                    promptNumber,
+                    0,
+                    false,
+                    null
+                );
+                transaction.Complete();
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        public string GetPromptNameForCourseAndPromptNumber(int customisationId, int promptNumber)
+        {
+            return courseAdminFieldsDataService.GetPromptNameForCourseAndPromptNumber(customisationId, promptNumber);
+        }
+
+        private CourseAdminFieldsResult? GetCourseCustomPromptsResultForCourse(
+            int customisationId,
+            int centreId,
+            int categoryId
+        )
+        {
+            var result = courseAdminFieldsDataService.GetCourseAdminFields(customisationId, centreId, categoryId);
```

Comment:
Should the data service filter out the fields with incorrect category id (as we're passing it in as a param)?

Reply:
I'm struggling to remember what categoryId is doing in there, I think I will just take it out